lib/src/Pos/Node/API.hs

```diff
@@ -547,47 +551,130 @@ instance ToSchema (V1 Version) where
         pure $ NamedSchema (Just "V1Version") $ mempty
             & type_ .~ SwaggerString
 
+
+newtype SecurityParameter = SecurityParameter Int
+    deriving (Show, Eq, Generic, ToJSON, FromJSON, Arbitrary)
+
+instance Buildable SecurityParameter where
+    build (SecurityParameter i) = bprint shown i
+
+instance ToSchema SecurityParameter where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "SecurityParameter") $ mempty
+            & type_ .~ SwaggerString
+
+
+instance ToSchema (V1 Core.SlotId) where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "Core.SlotId") $ mempty
+            & type_ .~ SwaggerString
+
+deriving instance ToJSON (V1 Core.SlotId)
+deriving instance FromJSON (V1 Core.SlotId)
+
+instance Arbitrary (V1 Core.SlotId) where
+    arbitrary = fmap V1 arbitrary
+
+
+instance ToSchema Byte where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "Byte") $ mempty
+            & type_ .~ SwaggerString
+
+instance Buildable Byte where
+    build b = bprint shown b
+
+
+instance Arbitrary (V1 Core.TxFeePolicy) where
+    arbitrary = fmap V1 arbitrary
+
+instance ToSchema (V1 Core.TxFeePolicy) where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "Core.TxFeePolicy") $ mempty
+            & type_ .~ SwaggerString
+
+deriving instance ToJSON (V1 Core.TxFeePolicy)
+deriving instance FromJSON (V1 Core.TxFeePolicy)
+
+
+instance Arbitrary (V1 Core.SlotCount) where
+    arbitrary = fmap V1 arbitrary
+
+instance ToSchema (V1 Core.SlotCount) where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "Core.SlotCount") $ mempty
+            & type_ .~ SwaggerString
+
+deriving instance ToJSON (V1 Core.SlotCount) -- TODO ?
+deriving instance FromJSON (V1 Core.SlotCount) --TODO ?
+
 -- | The @static@ settings for this wallet node. In particular, we could group
 -- here protocol-related settings like the slot duration, the transaction max size,
 -- the current software version running on the node, etc.
 data NodeSettings = NodeSettings
-    { setSlotDuration   :: !SlotDuration
-    , setSoftwareInfo   :: !(V1 Core.SoftwareVersion)
-    , setProjectVersion :: !(V1 Version)
-    , setGitRevision    :: !Text
+    { setSlotId            :: !(V1 Core.SlotId)
+    , setSlotDuration      :: !SlotDuration
+    , setSlotCount         :: !(V1 Core.SlotCount)
+    , setSoftwareInfo      :: !(V1 Core.SoftwareVersion)
+    , setProjectVersion    :: !(V1 Version)
+    , setGitRevision       :: !Text
+    , setMaxTxSize         :: !Byte
+    , setFeePolicy         :: !(V1 Core.TxFeePolicy)
+    , setSecurityParameter :: !SecurityParameter
```

Comment:
Same remark here, what's the unit of that thing :thinking: ?

Reply:
If anything: slots. But technically, no unit, I believe 🤔 

"k slots"
"k \in \mathbb{Z}"

https://cardanodocs.com/technical/
https://eprint.iacr.org/2016/889.pdf

Reply:
Well, `slots` sounds like a very valid unit to me then ^^"

Reply:
> Well, slots sounds like a very valid unit to me then ^^"

But… If `k = 2 slots` then `k slots = 2 slots^2` 😨

Are we looking past this and redefining (in a minor way) `k` for the sake of clarity in the API?

Reply:
I think this is the simplest form of my reasoning:

In the docs/paper we can find the expressions like `k: 60` (as in `k = 60`), and `k slots`.

Assume our API instead returns `k = 2 slots`, then:
```haskell
(let k = 2 slots  -- From our API
in k slots       -- we see this expression in the paper)

== 2 slots^2     -- this is absurd unless slots^2 == slots == 1, i.e has no dimension
```

Without a unit in our API we get this:
```haskell
(let k = 2       -- From our API
in k slots       -- we see this expression in the paper)

== 2 slots -- Very sane
```

**I'd be very happy to be wrong or to ignore this as this is fairly non-important in comparison to other things.**

Reasons I see I could be wrong:
- `k slots` in the text is meant more as "a distance d" rather than "d * meters"
- `slots = 1`
- We don't care about dimensional analysis here / it's unsuitable here ?
- ?

But maybe @paweljakubas could weigh in, who should know a lot more than me? Should `setSecurityParameter`, *k*, be described like `{"unit": "slots", quantity: 2}` or `2`  in the API?

Reply:
Actually — the simplest version of the argument is more trivial 😅: 

If the docs say `k = 60`, *k* doesn't have a dimension (and we can't say `k = 60 slots`).

Reply:
> we refer to k as let k = 2 in k slots, thus, the unit for securityParameter is slot. k refer to a number of slots... ?

@KtorZ I think you are saying that `securityParameter = k slots`. However, in the paper we can read: "some security parameter k", meaning `securityParameter = k`.